### ISSUE TITLE
Suggestion, make the plugin MUG compliant

### ISSUE THREAD

**Issue by smashyalts** [KaspianDev/AntiPopup#97]
The popup needs to be shown to be MUG compliant, hiding it is against the MUG (Minecraft usage guidelines) and therefor can get a server blacklisted, I suggest either removing the popup hiding completely or making it shown by default with a config option for people who don't care about following the game's rules.
![image](https://github.com/user-attachments/assets/472395a1-897c-4873-86a9-1ea986bcff88)
the MUG rule being broken
https://www.minecraft.net/en-us/usage-guidelines#commercial
![image](https://github.com/user-attachments/assets/59ac88e9-c94b-4e8f-9df1-bf3f51fcccd0)
![image](https://github.com/user-attachments/assets/7040e470-e0d3-4715-b512-9a13fe7e55a3)
Gamersafer
![image](https://github.com/user-attachments/assets/035884d9-6b20-4dc5-a0d7-363cc34eb288)
IP Justice (Mojang Enforcement Agent)
(keeping the chat signing disabled Is fine (for now))

**Comment by KaspianDev** [comment]
I don't see how a plugin that uses vanilla server features (no weird hacks are done, I'm only modifying existing packet property which is a part of the game) would be violating some weird rules.

Could you send me a part of the guidelines which specify how hiding the popup isn't allowed?

Also remember, AntiPopup itself isn't masking any behavior, client still sees that a chat message isn't signed by hovering on the left side of chat bar. Once you hover you will see a message explaining that the message is a "Server Message".

Blocking the popup is literally the same as toggling `enforce-secure-profile` but without any potential bad side-effects.

**Comment by KaspianDev** [comment]
tl;dr don't care

**Comment by smashyalts** [comment]
every message being seen as a server message and the popup are two different things, the popup (chat signing is disabled and therefor the warning needs to be there), if you could read the "proof" I included, chat signing (player messages sent as server messages) is "fine" (for now) but hiding the popup when it should be visible is against the MUG

**Comment by KaspianDev** [comment]
I am not changing that behavior, that's the core part of AntiPopup, hence the name.

If you hate your players and you wanna torture them with the popup on each join you can reenable popup in the config while keeping chat reporting disabled.

**Comment by smashyalts** [comment]
i dont hate my players but people dont know its against the MUG and keeping a plugin MUG non-compliant is not good, even if its the name or not

**Comment by KaspianDev** [comment]
I don't care about unrealistic rules to be honest.

Again, if mojang really wanted the popup they wouldn't give servers the authority to control it.
It's literally a feature in the game's protocol, designed that way on purpose.

**Comment by 0n** [comment]
🤓

**Comment by JerichoTorrent** [comment]
> I don't see how a plugin that uses vanilla server features (no weird hacks are done, I'm only modifying existing packet property which is a part of the game) would be violating some weird rules.
This is the same logic used to justify running a cracked server just because `offline-mode` is a setting in server.properties. It's still against the rules; point blank. Mojang Enforcement themselves has stated that your plugin violates the MUG. The right thing to do is AT LEAST put a note in your readme.md disclosing the fact that this plugin could get you in hot water and lead to your server getting blacklisted from authentication servers.

Your response clearly shows you care nothing for the people who use your plugin nor the community at large.

**Comment by 0n** [comment]
> > I don't see how a plugin that uses vanilla server features (no weird hacks are done, I'm only modifying existing packet property which is a part of the game) would be violating some weird rules.
> 
> This is the same logic used to justify running a cracked server just because `offline-mode` is a setting in server.properties. It's still against the rules; point blank. Mojang Enforcement themselves has stated that your plugin violates the MUG. The right thing to do is AT LEAST put a note in your readme.md disclosing the fact that this plugin could get you in hot water and lead to your server getting blacklisted from authentication servers.
> 
> Your response clearly shows you care nothing for the people who use your plugin nor the community at large.

🤓

**Comment by KaspianDev** [comment]
> > I don't see how a plugin that uses vanilla server features (no weird hacks are done, I'm only modifying existing packet property which is a part of the game) would be violating some weird rules.
> 
> This is the same logic used to justify running a cracked server just because `offline-mode` is a setting in server.properties. It's still against the rules; point blank. Mojang Enforcement themselves has stated that your plugin violates the MUG. The right thing to do is AT LEAST put a note in your readme.md disclosing the fact that this plugin could get you in hot water and lead to your server getting blacklisted from authentication servers.
> 
> Your response clearly shows you care nothing for the people who use your plugin nor the community at large.

1. Which part of the mug does offline mode break?
2. If you were following the conversation you'd know that antipopup has an option to reenable the popup.

**Comment by smashyalts** [comment]
Offline-mode isnt against the MUG, letting people without a legitimate account is (which is what offline-mode allows), the popup should be disabled by default and just let the config option be there to enable it again for people who don't care, and yes offline-mode servers arent actively being enforced on at this moment but will in the future, as confirmed by ip_justice, and also he was comparing "logic", just because its in the protocol (how else would the server sent to the client that info?), and you can manipulate it doesn't mean its following the MUG or allowed

**Comment by KaspianDev** [comment]
If you'd like to continue the discussion feel free to do so on my discord server.
